src/vehicle.cpp

```diff
@@ -3572,6 +3618,63 @@ void vehicle::power_parts()
         }
     }
 
+    if( has_part( "BOILER", true ) ) {
+        for( size_t b=0 ; b < boilers.size(); ++b ) {
+            printf("\nchecking boiler %zu\n", b);
+            if( !is_boiler_on( b ) ) {
+                printf("boiler is off\n");
+                continue;
+            }
+
+            printf("looking for tank\n");
+            const vehicle_part &pt = parts[boilers[b]];
+            auto tank = std::find_if( parts.begin(), parts.end(),
+                      [&pt](const vehicle_part &e){
+                          return pt.mount == e.mount && e.is_tank() && \
+                          e.can_reload("steam");
+                      });
+            if( tank == parts.end() ) {
+                printf("skipping boiler due to lack of tank space\n");
+                continue;
+            }
+
+            //calculate tank space avaliable
+            const float space_left = tank->ammo_capacity() - tank->ammo_remaining();
+
+            if(space_left < 1700){
```

Comment:
This `1700` magic constant really needs to be extracted somewhere and either jsonized, acquired from an item property (most likely from steam item's volume) or documented.

Reply:
It is from wikipedia. Steam has roughly 1700 times the volume of the water it came from.

Second paragraph first sentence. https://en.wikipedia.org/wiki/Steam

Reply:
Then use a clearly named or commented constant, not 1700 directly. For example `water_to_steam_volume`